iModelCore/ECPresentation/Tests/NonPublished/Unit/Queries/CustomFunctionTests.cpp

```diff
@@ -169,6 +169,33 @@ TEST_F(CustomFunctionTests, GetECInstanceDisplayLabel_UsesInstanceLabel)
     ASSERT_STREQ(GetDisplayLabelJson("CustomLabel").c_str(), stmt.GetValueText(0));
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST_F(CustomFunctionTests, GetECInstanceDisplayLabel_UsesInstanceLabelAndFormats)
+    {
+    ECClassCP classJ = s_project->GetECDb().Schemas().GetClass("RulesEngineTest", "ClassJ");
+    IECInstancePtr instanceJ = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classJ, [](IECInstanceR instance) {instance.SetValue("DisplayLabel", ECValue("CustomLabel")); });
+    ECInstanceId instanceJId;
+    ECInstanceId::FromString(instanceJId, instanceJ->GetInstanceId().c_str());
+
+    TestPropertyFormatter propertyFormatter;
+    propertyFormatter.SetValueFormatter([](Utf8StringR formattedValue, ECPropertyCR property, ECValueCR value, ECPresentation::UnitSystem)
+        {
+        formattedValue = "_" + property.GetAsPrimitiveProperty()->GetDisplayLabel() + "_";
+        return SUCCESS;
+        });
+
+    CustomFunctionsContext ctx(*m_schemaHelper, m_connections, *m_connection, m_ruleset->GetRuleSetId(), *m_rulesPreprocessor, m_rulesetVariables, nullptr, m_schemaHelper->GetECExpressionsCache(), m_nodesFactory, nullptr, nullptr, nullptr, &propertyFormatter);
+
+    ECSqlStatement stmt;
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.Prepare(GetDb(), "SELECT GetECInstanceDisplayLabel(?, ?, DisplayLabel, '') FROM RET.ClassJ"));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindId(1, classJ->GetId()));
+    ASSERT_TRUE(ECSqlStatus::Success == stmt.BindId(2, instanceJId));
+    ASSERT_TRUE(DbResult::BE_SQLITE_ROW == stmt.Step());
+    ASSERT_STREQ(GetDisplayLabelJson("_CustomLabel_").c_str(), stmt.GetValueText(0));

```

Comment:
This should take care of the test failure

```suggestion
    ASSERT_STREQ(GetDisplayLabelJson("CustomLabel", "_CustomLabel_").c_str(), stmt.GetValueText(0));
```